### ISSUE TITLE
Microsoft.RecoveryServices/vaults/backupPolicies

### ISSUE THREAD

**Issue by slavizh** [Azure/arm-template-whatif#97]
## Describe the noise

**Resource type** (i.e. Microsoft.Storage/storageAccounts)
Microsoft.RecoveryServices/vaults/backupPolicies

**apiVersion** (i.e. 2019-04-01)
2019-05-13

**Client (PowerShell, Azure CLI, or API)**
PowerShell

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**
No change at all on the policy

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**

```
  ~ Microsoft.RecoveryServices/vaults/acc-sentia-vault/backupPolicies/Test-FS-policy [2019-05-13]
    - properties.protectedItemsCount:                    0
    - properties.schedulePolicy.scheduleWeeklyFrequency: 0
    + location:                                          "West Europe"
    + properties.WorkloadType:                           "AzureFileShare"
    + tags:

        application:     "bkp"
        costCenter:      "IT"
        department:      "IT"
        environment:     "Test"
        productName:     "backup-management"
        solutionVersion: "1.1.0"



```

**Additional context**

**Comment by alex-frankel** [comment]
for `tags`, `location` and `properties.WorkloadType`, it looks like the RP is dropping those properties and this resource does not actually support them. Can you double check if those properties are necessary? It is likely the same issue as #96

**Comment by slavizh** [comment]
Both tags and location are listed in the official docs: https://docs.microsoft.com/en-us/azure/templates/microsoft.recoveryservices/2016-12-01/vaults/backuppolicies 

About WorkloadType - If it is not needed why when you create policy via the Portal (which creates ARM Template) that property is listed? 

![image](https://user-images.githubusercontent.com/11890856/87144660-c8f8c280-c2b0-11ea-9489-0fffb9af2cca.png)


I would say all the problems are within the Vaults API as their APIs are one of the worst. Just look if you can put File Share under backup with ARM Template (spoiler - you cannot) :)

**Comment by shenglol** [comment]
I fixed the first two properties that are shown as Delete. `properties.protectedItemsCount` is read-only and `properties.schedulePolicy.scheduleWeeklyFrequency` has a default value `0` when `properties.schedulePolicy.schedulePolicyType` is `"SimpleSchedulePolicy"`.

As for the created ones, I confirmed that `tags` and `location` are not supported, though they are declared in Swagger. `properties.workloadType` is legit and is required when creating the resource. However, the RP doesn't return it upon GET, which appears to be a provider bug to me.

>I would say all the problems are within the Vaults API as their APIs are one of the worst

Totally agree...
